src/gtc/passes/oir_optimizations/horizontal_execution_merging.py

```diff
@@ -23,6 +23,44 @@
 from .utils import AccessCollector, symbol_name_creator
 
 
+class NameUniqifier(NodeTranslator):
+    @dataclass
+    class Context:
+        used_names: Set[str]
+
+    @classmethod
+    def apply(
+        cls, vertical_loops: List[oir.VerticalLoop], skip_names: Set[str]
+    ) -> List[oir.VerticalLoop]:
+        return cls().visit(vertical_loops, ctx=cls.Context(used_names=skip_names))
+
+    def visit_HorizontalExecution(
+        self, node: oir.HorizontalExecution, *, ctx: "Context"
+    ) -> oir.HorizontalExecution:
+        new_symbol_name = symbol_name_creator(ctx.used_names)
+        name_map = {
+            d.name: new_symbol_name(d.name) for d in node.declarations if d.name in ctx.used_names
+        }
+
+        ctx.used_names |= {d.name for d in node.declarations}
```

Comment:
shouldn't this add the new names instead of the old ones?

Reply:
Good catch! It should have a combination of the new (if collision in used_names) and old names (if no collision detected). I fixed this.